src/System.IO.Ports/tests/Legacy/SerialPort/Stress01.cs

```diff
@@ -6,159 +6,96 @@
-        com2.Encoding = System.Text.Encoding.Unicode;
-
-        com1.BaudRate = 115200;
-        com2.BaudRate = 115200;
+            com1.BaudRate = 115200;
+            com2.BaudRate = 115200;
 
-        com1.Open();
+            com1.Open();
 
-        if (!com2.IsOpen) //This is necessary since com1 and com2 might be the same port if we are using a loopback
-            com2.Open();
+            if (!com2.IsOpen) //This is necessary since com1 and com2 might be the same port if we are using a loopback
+                com2.Open();
 
-        sw.Start();
-        while (sw.ElapsedMilliseconds < MAX_RUN_TIME)
-        {
-            switch (random.Next(0, 2))
+            sw.Start();
+            while (sw.ElapsedMilliseconds < MAX_RUN_TIME)
             {
-                case 0: //Write
-                    if (com2.BytesToRead < MAX_BUFFER_SIZE)
-                    {
-                        int maxNumberOfCharactes = (MAX_BUFFER_SIZE - com2.BytesToRead) / 2;
-                        int numberOfCharacters = random.Next(0, Math.Min(xmitCharBuffer.Length, maxNumberOfCharactes) + 1);
-                        int expectedBytesToRead = com2.BytesToRead + 2 * numberOfCharacters;
-
-                        //						Console.WriteLine("Writing {0,5} characters BytesToRead={1,5}", numberOfCharacters, com2.BytesToRead);
-                        com1.Write(xmitCharBuffer, 0, numberOfCharacters);
-                        buffer.Append(xmitCharBuffer, 0, numberOfCharacters);
-
-                        retValue &= TCSupport.WaitForPredicate(delegate () { return com2.BytesToRead == expectedBytesToRead; }, 60000,
-                            "Err_29829haie Expected to received {0} bytes actual={1}", expectedBytesToRead, com2.BytesToRead);
-                    }
-                    break;
-                case 1: //Read
-                    if (0 < com2.BytesToRead)
-                    {
-                        int maxNumberOfCharactes = com2.BytesToRead / 2;
-                        int numberOfCharacters = random.Next(0, Math.Min(rcvCharBuffer.Length, maxNumberOfCharactes) + 1);
-                        int actualNumberOfCharactersRead;
-                        int expectedBytesToRead = com2.BytesToRead - (2 * numberOfCharacters);
-
-                        //						Console.WriteLine("Reading {0,5} characters BytesToRead={1,5}", numberOfCharacters, com2.BytesToRead);
-                        actualNumberOfCharactersRead = com2.Read(rcvCharBuffer, 0, numberOfCharacters);
-
-                        if (actualNumberOfCharactersRead == numberOfCharacters)
+                switch (random.Next(0, 2))
+                {
+                    case 0: //Write
+                        if (com2.BytesToRead < MAX_BUFFER_SIZE)
                         {
-                            retValue &= buffer.CompareAndRemove(rcvCharBuffer, 0, numberOfCharacters);
+                            int maxNumberOfCharactes = (MAX_BUFFER_SIZE - com2.BytesToRead) / 2;
```

Comment:
:) fair enough. This is typical of those old desktop tests. I was also going to comment on the wierd prefixes they put on the error messages, but maybe we will be glad later when one of them fires and for some reason there is no line number.